src/Gql.js

```diff
@@ -0,0 +1,215 @@
+import _ from 'lodash';
+import {parse, validate, execute} from 'graphql';
+
+import {formatError} from './errors';
+import {makeSchemaFromConfig} from './makeSchemaFrom';
+
+const ONE_DAY = 24 * 3600 * 1000;
+const ENUM_PREFIX = '__ENUM__::';
+// const VAR_PREFIX = '__VAR__::';
+
+// const NO_QUOTES_REGEX = new RegExp(`^(${ENUM_PREFIX}|${VAR_PREFIX})([A-Za-z_]+)$`);
+const ENUM_REGEX = new RegExp(`^${ENUM_PREFIX}([A-Za-z_]+)$`);
+
+function graphql(schemaObj, query, rootValue, contextValue, variableValues) {
+	// we are not using the inbuilt graphql function because it validates
+	// the graphql, which is an expensive operation
+	// return graphql(schema, query, rootValue, contextValue, variableValues);
+	// taken from: https://github.com/graphql/graphql-js/blob/master/src/graphql.js
+
+	// parse
+	let document;
+	try {
+		document = parse(query);
+	}
+	catch (syntaxError) {
+		return Promise.resolve({errors: [syntaxError]});
+	}
+
+	// validate only in developement
+	if (process.env.NODE_ENV !== 'production') {
```

Comment:
this could also be set through options?

Reply:
you mean an option of `validateGraphql: boolean` in `Gql.exec`?

Reply:
yea, or through constructor as this would not be valid for `api` exec